runtime/nvqir/CircuitSimulator.h

```diff
@@ -843,20 +838,25 @@ class CircuitSimulatorBase : public CircuitSimulator {
   }
 
   /// @brief Deallocate the qubit with give index
-  void deallocate(const std::size_t qubitIdx) override {
+  void deallocateQudit(const std::size_t uid) override {
```

Comment:
Nit: maybe it's just me, but I think that `qubits` or `qudits` is a bit more descriptive than `uids`. I'm curious what the "u" in `uids` represents.